operators/src/processing/new_expression/query_processor.rs

```diff
@@ -0,0 +1,401 @@
+use std::{marker::PhantomData, sync::Arc};
+
+use async_trait::async_trait;
+use futures::{stream::BoxStream, try_join, StreamExt, TryStreamExt};
+use geoengine_datatypes::{
+    primitives::{SpatialPartition2D, TimeInterval},
+    raster::{
+        GeoTransform, Grid2D, GridIdx2D, GridShape2D, GridShapeAccess, NoDataValue, Pixel,
+        RasterTile2D,
+    },
+};
+use num_traits::AsPrimitive;
+use rayon::iter::{IntoParallelIterator, IntoParallelRefIterator, ParallelIterator};
+
+use crate::{
+    engine::{BoxRasterQueryProcessor, QueryContext, QueryProcessor, RasterQueryRectangle},
+    util::{stream_zip::StreamTupleZip, Result},
+};
+
+use super::compiled::LinkedExpression;
+
+pub struct ExpressionQueryProcessor<TO, Sources>
+where
+    TO: Pixel,
+{
+    pub sources: Sources,
+    pub phantom_data: PhantomData<TO>,
+    pub program: Arc<LinkedExpression>,
+    pub no_data_value: TO,
```

Comment:
if everything is turned into f64, do we need this generic?

Reply:
We output a Raster of type `TO`, so I guess this makes sense.